src/Core/Repository/UserRepository.php

```diff
@@ -2,12 +2,41 @@
 
 namespace Forumify\Core\Repository;
 
+use Doctrine\Persistence\ManagerRegistry;
 use Forumify\Core\Entity\User;
+use Symfony\Bridge\Doctrine\Security\User\UserLoaderInterface;
 
-class UserRepository extends AbstractRepository
+class UserRepository extends AbstractRepository implements UserLoaderInterface
 {
+    private $settingRepository;
+
+    public function __construct(ManagerRegistry $registry, SettingRepository $settingRepository)
+    {
+        parent::__construct($registry);
+        $this->settingRepository = $settingRepository;
+    }
+
     public static function getEntityClass(): string
     {
         return User::class;
     }
+
+    public function loadUserByIdentifier(string $usernameOrEmail): ?User
+    {
+        $queryBuilder = $this->createQueryBuilder('u');
+
+        if ($this->settingRepository->get('core.enable_email_login') == null or $this->settingRepository->get('core.enable_email_login') == "username") {
+            $queryBuilder->andWhere('u.username = :query');
+        } elseif ($this->settingRepository->get('core.enable_email_login') == "email") {
+            $queryBuilder->andWhere('u.email = :query');
+        } elseif ($this->settingRepository->get('core.enable_email_login') == "both") {
```

Comment:
Same here as in the comment below in twig regarding the check for username or null to just be an `else` at the bottom. Let's say someone sets it to 'blipblop', then this if would not add any where's, which would result in the first user in the database being returned. Usually this is the super admin, so every user would log in as super user 😅